internal/goro/group.go

```diff
@@ -0,0 +1,77 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package goro
+
+import (
+	"context"
+	"sync"
+)
+
+// Group manages a set of long-running goroutines as a unit. Group differs from
+// `errgroup.Group` in that it does not link the lifetimes of the spawned
+// goroutines and it is not expected that a client will call Wait() without
+// first calling Cancel(). If you're looking for a short-lived (e.g.
+// request-scoped) group of transient goroutines, you probably want
+// `errgroup.Group`. The expectation is that an instance of this type will be
+// held as a member field of an object that has standard Start()/Stop()
+// semantics (e.g. a `common.Daemon`). The zero-value of this type is valid.
```

Comment:
I think mentioning `errgroup` here makes it more confusing. What's about simple example on how it should be used instead? You have few in your tests.

Reply:
I put it here because it would have been my own first question: why not errgroup.Group? But since I've subsequently added a package comment (#1900), I think I'll move that reference there.

Reply:
Oh and I also add an example that will show up in godoc in that PR